build/DownloadCFamilyPlugin/Common.cs

```diff
@@ -0,0 +1,239 @@
+            return fileName;
+        }
+
+        public static void UnzipJar(string jarFilePath, string destinationFolder, TaskLoggingHelper logger)
+        {
+            if (Directory.GetFiles(destinationFolder).Length > 1)
+            {
+                // Unzipping the jar is slow so skip if possible
+                LogMessage($"Skipping unzipping the jar because the destination folder already contains multiple files.", logger);
+                return;
+            }
+
+            LogMessage($"Unzipping jar file...", logger);
+            var timer = Stopwatch.StartNew();
+
+            ZipFile.ExtractToDirectory(jarFilePath, destinationFolder);
+
+            timer.Stop();
+            LogElapsedTime("Unzipped jar file", timer, logger);
+        }
+
+        public static void UncompressAndUnzipTgx(string tarFilePath, string destinationFolder, TaskLoggingHelper logger)
+        {
+            // The txz file is compressed using XZ compression and zipped using the tar format.
+            // There is no built-in framework support for these format so we are using two
+            // open source libraries, both licensed under the MIT license.
+            Common.EnsureWorkingDirectoryExist(destinationFolder, logger);
+
+            var uncompresssedFile = Common.DecompressXZFile(tarFilePath, destinationFolder, logger);
+            Common.ExtractTarToDirectory(uncompresssedFile, destinationFolder, logger);
+        }
+
+        public static string DecompressXZFile(string sourceFilePath, string destinationDirectory, TaskLoggingHelper logger)
+        {
+            var destFile = Path.Combine(destinationDirectory, Path.GetFileName(sourceFilePath)) + ".uncompressed";
+            if (File.Exists(destFile))
+            {
+                LogMessage($"Uncompressed tar file already exists: {destFile}", logger);
+                return destFile;
+            }
+
+            using (Stream xz = new XZStream(File.OpenRead(sourceFilePath)))
+            using (Stream outputStream = new FileStream(destFile, FileMode.CreateNew))
+            {
+                xz.CopyTo(outputStream);
+            }
+            return destFile;
+        }
+
+        public static void UncompressAndUnzipTgz(string tarFilePath, string destinationFolder, TaskLoggingHelper logger)
+        {
+            // The txz file is compressed using XZ compression and zipped using the tar format.
+            // There is no built-in framework support for these format so we are using two
+            // open source libraries, both licensed under the MIT license.
+            EnsureWorkingDirectoryExist(destinationFolder, logger);
+
+            var uncompresssedFile = DecompressGZipFile(tarFilePath, destinationFolder, logger);
+            ExtractTarToDirectory(uncompresssedFile, destinationFolder, logger);
+        }
+
+        public static string DecompressGZipFile(string sourceFilePath, string destinationDirectory, TaskLoggingHelper logger)
```

Comment:
I see that this is new too, is that what we will need for eslint? why didn't the existing DecompressXZFile work?

Reply:
Ah. This addition is from the JS prototype version.

I originally created this file as part of the prototype, which is when I originally extracted to the code. I used that as the basis of this new file, so a couple of extra changes have sneaked in.

The CFamily and TypeScript files use different formats.
CFamily use `.tgx` files, TypeScript uses `.tgz`